### PR TITLE
feat: support `next@15`

### PR DESCRIPTION
**Pull request by blechatellier**
This PR updates `next-international` to support async changes in `next@15`.

### CODE REVIEW

**Comment by blechatellier** [comment]
@QuiiBz hey, would appreciate a review when available to be able to upgrade to next@15

**Comment by fstodulski** [comment]
☝️ I there a possibility to merge it ASAP?

File: examples/next-app/package.json
Comment (QuiiBz) [inline examples/next-app/package.json:1]: This example needs to be updated, as the page is accessing `params.locale` without `await`ing it.

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks for the PR and sorry for the delay, I'm travelling away for a few days.

Since Next.js 15 contains several breaking changes to dynamic APIs and `next-international`'s v2 version is still [wip](https://github.com/QuiiBz/next-international/pull/361) (I'm planning to work on it soon), we need to support both Next.js 14 and 15 on `next-international@1.x.x`. It might already just work with those changes in both versions given JavaScript allows non-promise values to be `await`'ed, but we need to double-check.

I left a comment for the `next-app` example that is currently broken, and the documentation also needs to be updated. I'll be able to do it on Monday once I'm back home, but feel free to update the PR before if you can so we can support Next.js 15 asap.

**Comment by blechatellier** [comment]
@QuiiBz thanks, I've updated the docs and example.

**Review by QuiiBz — APPROVED** [review]
Updated the docs to still show how to use next-international with Next.js < 15, but the default will be with the new `await`able APIs.

I'm releasing `next-international@1.3.0` shortly after merging this PR, thanks!

**Comment by blechatellier** [comment]
Thanks @QuiiBz

**Comment by mxzinke** [comment]
@QuiiBz @blechatellier 

Did you miss the type safety for `getCurrentLocale`, since this now returns a promise?

**Pull request merged**